gulpfile.babel.js

```diff
@@ -0,0 +1,190 @@
+  config.dir.src.templates = `${config.sourceRoot}/_patterns`;
+  config.dir.src.js = `${config.sourceRoot}/js`;
+
+  config.dir.test.sass = `${config.testRoot}/sass`;
+
+  config.dir.out.css = `${config.exportRoot}/css`;
+  config.dir.out.sass = `${config.dir.out.css}/sass`;
+  config.dir.out.images = `${config.exportRoot}/images`;
+  config.dir.out.fonts = `${config.exportRoot}/fonts`;
+  config.dir.out.templates = `${config.exportRoot}/templates`;
+
+  config.files = {
+    src: {},
+    test: {},
+    out: {},
+  };
+  config.files.src.css = [
+    `${config.dir.src.css}/**/*.css`,
+    `${config.dir.src.css}/**/*.map`,
+    `!${config.dir.src.css}/pattern-scaffolding.css`,
+  ];
+  config.files.src.sass = `${config.dir.src.sass}/**/*.scss`;
+  config.files.src.sassEntryPoint = `${config.dir.src.sass}/${invocationOptions.sassEntryPoint}`;
+  config.files.src.images = [`${config.dir.src.images}/*`, `${config.dir.src.images}/**/*`];
+  config.files.src.fonts = [`${config.dir.src.fonts}/*`, `${config.dir.src.fonts}/**/*`];
+  config.files.src.templates = [`${config.dir.src.templates}/*.twig`, `${config.dir.src.templates}/**/*.twig`];
+  config.files.src.derivedConfigs = [
+    `${config.dir.src.sass}/variables/**/*`,
+    `${config.dir.src.js}/derivedConfig.json`,
+  ];
+
+  config.files.test.sass = `${config.dir.test.sass}/**/*.spec.scss`;
+  config.files.test.sassTestsEntryPoint = `${config.dir.test.sass}/test_sass.js`;
+
+  config.files.out.cssFilename = invocationOptions.cssOutFilename;
+
+  return config;
+
+};
+
+const config = buildConfig(process.argv, 'source', 'test', 'export');
+
+// Builders
+
+const cleanSharedConfig = () => del(config.files.src.derivedConfigs);
+
+export const distributeSharedConfig = gulp.series(cleanSharedConfig, distributeConfig);
+
+const cleanCss = () => del(config.files.src.css);
+
+export const lintSass = () => {
+  if (!config.sassLinting) {
+    console.info('Skipping sass:lint');
+    return Promise.resolve();
+  }
+
+  const processors = [
+    stylelint(),
+    reporter(
+      {
+        clearMessages: true,
+        throwError: true,
+      },
+    ),
+  ];
+
+  return gulp.src([config.files.src.sass])
+    .pipe(postcss(processors, {syntax: syntaxScss}));
+};
+
+export const testSass = () =>
+  gulp.src(config.files.test.sassTestsEntryPoint)
+    .pipe(mocha({reporter: 'spec'}));
+
+export const generateCss = () => {
+  const sassOptions = config.environment === 'production' ? {outputStyle: 'compressed'} : null;
+  return gulp.src(config.files.src.sassEntryPoint)
+    .pipe(sourcemaps.init())
+    .pipe(sassGlob())
+    .pipe(sass(sassOptions).on('error', sass.logError))
+    .pipe(replace(/\.\.\/\.\.\/fonts\//g, '../fonts/'))
+    .pipe(rename(config.files.out.cssFilename))
+    .pipe(sourcemaps.write('./'))
+    .pipe(gulp.dest(config.dir.src.css));
+};
+
+export const build = gulp.parallel(
+  lintSass,
+  testSass,
+  gulp.series(cleanCss, generateCss),
```

Comment:
This is different in that the linted, testing and compiling all happen in parallel, rather than compiling happening after the others.